### PR TITLE
doc: release-notes-latest: Summarize networking changes from the upmerge

### PR DESCRIPTION
**Pull request by rlubos**
Filtered out only relevant ones for us.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/releases/release-notes-latest.rst [hunk 1]
```diff
@@ -24,7 +24,29 @@ The Zephyr fork in |NCS| contains all commits from the upstream Zephyr repositor
 
 The following list summarizes the most important changes inherited from upstream Zephyr:
 
-*
+* LwM2M:
+
+  - Fixed a bug, where FOTA socket was not closed after the download (PULL mode).
```
Comment (ru-fu) [inline doc/nrf/releases/release-notes-latest.rst:29]:
```suggestion
  * Fixed a bug where a FOTA socket was not closed after the download (PULL mode).
```

Comment (ru-fu) [reply doc/nrf/releases/release-notes-latest.rst:29]: Please replace all "-" with "*".

File: doc/nrf/releases/release-notes-latest.rst [hunk 2]
```diff
@@ -24,7 +24,29 @@ The Zephyr fork in |NCS| contains all commits from the upstream Zephyr repositor
...
+  - Allow to specify how early before the timeout shall Register Update be sent.
```
Comment (ru-fu) [inline doc/nrf/releases/release-notes-latest.rst:30]:
```suggestion
  - Allowed to specify how early before the time-out the Register Update will be sent.
```

Comment (ru-fu) [reply doc/nrf/releases/release-notes-latest.rst:30]: How is this specified? Is it a Kconfig option, or a parameter to a function?

File: doc/nrf/releases/release-notes-latest.rst [hunk 3]
```diff
@@ -24,7 +24,29 @@ The Zephyr fork in |NCS| contains all commits from the upstream Zephyr repositor
...
+  - Added ObjLnk resource type support.
+
+* MQTT:
+
+  - `utf8` pointer in `mqtt_utf8` struct is now const.
```
Comment (ru-fu) [inline doc/nrf/releases/release-notes-latest.rst:35]:
```suggestion
  - The ``utf8`` pointer in the :c:type:`mqtt_utf8` struct is now const.
```
It should be possible to link to the struct (let's see if the CI agrees 😉  ).

File: doc/nrf/releases/release-notes-latest.rst [hunk 4]
```diff
@@ -24,7 +24,29 @@ The Zephyr fork in |NCS| contains all commits from the upstream Zephyr repositor
...
+  - The default `clean_session` value is now configurable with Kconfig.
```
Comment (ru-fu) [inline doc/nrf/releases/release-notes-latest.rst:36]:
```suggestion
  - The default ``clean_session`` value is now configurable with Kconfig (see :option:`CONFIG_MQTT_CLEAN_SESSION`).
```

File: doc/nrf/releases/release-notes-latest.rst [hunk 5]
```diff
@@ -24,7 +24,29 @@ The Zephyr fork in |NCS| contains all commits from the upstream Zephyr repositor
...
+
+* OpenThread:
```
Comment (ru-fu) [inline doc/nrf/releases/release-notes-latest.rst:38]: I think you need an empty line after this.

File: doc/nrf/releases/release-notes-latest.rst [hunk 6]
```diff
@@ -24,7 +24,29 @@ The Zephyr fork in |NCS| contains all commits from the upstream Zephyr repositor
...
+  - Updated OpenThread revision up to upstream commit e653478c503d5b13207b01938fa1fa494a8b87d3.
+  - Implemented missing `enable` API function for OpenThread interface.
+  - Cleaned up OpenThread Kconfig file - OpenThread dependencies are now enabled automatically.
+  - Allow the applicaiton to register OpenThread state change callback.
+  - Reimplemented logger glue layer, for better performance.
+  - OpenThread thread priority class is now configurable.
+  - Several new Kconfig options added, customizing OT stack.
```
Comment (ru-fu) [inline doc/nrf/releases/release-notes-latest.rst:45]:
```suggestion
  * Updated the OpenThread revision to upstream commit e653478c503d5b13207b01938fa1fa494a8b87d3.
  * Implemented a missing ``enable`` API function for the OpenThread interface.
  * Cleaned up the OpenThread Kconfig file.
    OpenThread dependencies are now enabled automatically.
  * Allowed the application to register a callback function for OpenThread state changes.
  * Reimplemented the logger glue layer for better performance.
  * Updated the OpenThread thread priority class to be configurable.
  * Added several Kconfig options to customize the OpenThread stack.
```

Comment (ru-fu) [inline doc/nrf/releases/release-notes-latest.rst:45]: Can we list the Kconfig options that were added, or are there too many?

File: doc/nrf/releases/release-notes-latest.rst [hunk 7]
```diff
@@ -24,7 +24,29 @@ The Zephyr fork in |NCS| contains all commits from the upstream Zephyr repositor
...
+
+* Socket offloading:
+
+ - Removed dependency to `POSIX_NAMES` config.
```
Comment (ru-fu) [inline doc/nrf/releases/release-notes-latest.rst:49]:
```suggestion
 * Removed dependency to the :option:`CONFIG_NET_SOCKETS_POSIX_NAMES` configuration option.
```
Is that the correct one?

Comment (rlubos) [reply doc/nrf/releases/release-notes-latest.rst]: Yes, that's correct.

File: doc/nrf/releases/release-notes-latest.rst (hunk 2 above)
Comment (rlubos) [reply doc/nrf/releases/release-notes-latest.rst]: It's a Kconfig option.

File: doc/nrf/releases/release-notes-latest.rst (hunk 6 above)
Comment (rlubos) [reply doc/nrf/releases/release-notes-latest.rst]: There were quite a few of them, IMO not worth listing them all.

File: doc/nrf/releases/release-notes-latest.rst [hunk 8]
```diff
@@ -24,7 +24,31 @@ The Zephyr fork in |NCS| contains all commits from the upstream Zephyr repositor
 
 The following list summarizes the most important changes inherited from upstream Zephyr:
 
-*
+* LwM2M:
+
+  * Fixed a bug where a FOTA socket was not closed after the download (PULL mode).
+  * Allowed to specify how early before the time-out the Register Update will be sent.
```
Comment (ru-fu) [inline doc/nrf/releases/release-notes-latest.rst:30]:
```suggestion
  * Added a Kconfig option :option:`CONFIG_LWM2M_SECONDS_TO_UPDATE_EARLY` that specifies how long before the time-out the Registration Update will be sent.
```

File: doc/nrf/releases/release-notes-latest.rst [hunk 9]
```diff
@@ -24,7 +24,31 @@ The Zephyr fork in |NCS| contains all commits from the upstream Zephyr repositor
 
 The following list summarizes the most important changes inherited from upstream Zephyr:
 
-*
+* LwM2M:
```
Comment (carlescufi) [inline doc/nrf/releases/release-notes-latest.rst:27]: Can you move this whole section under a "Networking" header?

Comment (rlubos) [reply doc/nrf/releases/release-notes-latest.rst:30]: Yes, done

File: doc/nrf/releases/release-notes-latest.rst [hunk 10]
```diff
@@ -30,6 +30,34 @@ The following list summarizes the most important changes inherited from upstream
     The kernel now uses the standard Zephyr logging API at DBG log level for this purpose.
     The logging module used for the kernel is named ``os``.
 
+* Networking
```
Comment (ru-fu) [inline doc/nrf/releases/release-notes-latest.rst:33]:
```suggestion
* Networking:
```

**Pull request merged**